src/gui/networksettings.cpp

```diff
@@ -195,6 +196,18 @@ void NetworkSettings::loadMeteredSettings()
     _ui->pauseSyncWhenMeteredCheckbox->setVisible(false);
 }
 
+void NetworkSettings::loadCaptivePortalSettings()
+{
+    if (QNetworkInformation *qNetInfo = QNetworkInformation::instance()) {
+        if (qNetInfo->supports(QNetworkInformation::Feature::CaptivePortal)) {
+            _ui->pauseSyncWhenBehindCaptivePortalCheckBox->setChecked(ConfigFile().pauseSyncWhenBehindCaptivePortal());
+            return;
+        }
+    }
+
+    _ui->pauseSyncWhenBehindCaptivePortalCheckBox->setVisible(false);
```

Comment:
We need to hide the tls dialog if enabled, I'd also say we need to pause, as util the captive portal is accepted there is no way to sync.